logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -73,7 +80,7 @@ def close
       end
 
       def empty?
-        true # synchronous queue is alway empty
+        @queue.queue.isEmpty
```

Comment:
Nevermind. Github's diff rendering made this hard to understand. I didn't see this was part of ReadClient and it looked part of WrappedSynchronousQueue..

Ignore this comment.